corehq/apps/domain/forms.py

```diff
@@ -1157,3 +1167,430 @@ def process_submission(self, domain=None):
+    )
+
+    emails = forms.CharField(
+        help_text=ugettext_noop(
+            'This is who will receive invoices if the Client exceeds the user '
+            'or SMS limits in their plan.'
+        ),
+        label=ugettext_noop('Partner Contact Emails'),
+        max_length=BillingContactInfo._meta.get_field('emails').max_length,
+    )
+
+    start_date = forms.DateField(
+        help_text=ugettext_noop('Date the project needs access to features.'),
+        label=ugettext_noop('Start Date'),
+    )
+
+    end_date = forms.DateField(
+        help_text=ugettext_noop(
+            '1 year after the deployment date (date the project goes live).'
+        ),
+        label=ugettext_noop('End Date'),
+    )
+
+    sms_credits = forms.DecimalField(
+        initial=0,
+        label=ugettext_noop('SMS Credits'),
+    )
+
+    user_credits = forms.IntegerField(
+        initial=0,
+        label=ugettext_noop('User Credits'),
+    )
+
+    def __init__(self, domain, web_user, *args, **kwargs):
+        super(ContractedPartnerForm, self).__init__(domain, web_user, *args, **kwargs)
+
+        self.helper = FormHelper()
+        self.helper.form_class = 'form-horizontal'
+
+        plan_edition = self.current_subscription.plan_version.plan.edition if self.current_subscription else None
+        if plan_edition not in [
+            first for first, second in self.fields['software_plan_edition'].choices
+        ]:
+            self.fields['start_date'].initial = datetime.date.today()
+            self.fields['end_date'].initial = datetime.date.today() + relativedelta(years=1)
+            self.helper.layout = crispy.Layout(
+                crispy.Field('software_plan_edition'),
+                crispy.Field('fogbugz_client_name'),
+                crispy.Field('emails', css_class='input-xxlarge'),
+                crispy.Field('start_date', css_class='date-picker'),
+                crispy.Field('end_date', css_class='date-picker'),
+                crispy.Field('sms_credits'),
+                crispy.Field('user_credits'),
+                crispy.HTML(_(
+                    '<p><i class="icon-info-sign"></i> Clicking "Update" will set '
+                    'up the subscription in CommCareHQ to one of our standard '
+                    'contracted plans.  If you need to set up a non-standard plan, '
+                    'please email accounts@dimagi.com.</p>'
+                )),
+                self.form_actions
+            )
+        else:
+            self.fields['fogbugz_client_name'].initial = self.current_subscription.account.name
+            self.fields['emails'].initial = self.current_subscription.account.billingcontactinfo.emails
+            self.fields['end_date'].initial = self.current_subscription.date_end
+            self.helper.layout = crispy.Layout(
+                TextField('software_plan_edition', plan_edition),
+                crispy.Hidden('software_plan_edition', plan_edition),
```

Comment:
```TextField``` is just text on the screen, the ```Hidden``` field enables that data to be sent in the form submission